### PR TITLE
OSF: Used model.NewPointer everywhere

### PR DESCRIPTION
**Pull request by agnivade**
```release-note
NONE
```

### CODE REVIEW

File: server/public/model/builtin.go [hunk 1]
```diff
@@ -3,11 +3,14 @@
 
 package model
 
+// These functions are deprecated. Please use only NewPointer.
 func NewBool(b bool) *bool       { return &b }
 func NewInt(n int) *int          { return &n }
 func NewInt64(n int64) *int64    { return &n }
 func NewString(s string) *string { return &s }
-func NewPointer[T any](t T) *T   { return &t }
```
Comment (agarciamontoro) [inline server/public/model/builtin.go:10]: When will we remove these?

**Review by agarciamontoro — APPROVED** [review]
Great, thanks!

File: server/public/model/builtin.go (hunk 1 above)
Comment (agnivade) [reply server/public/model/builtin.go:10]: Probably with the next ESR. Technically these could go away right now, but just wanted to give a bit of leeway to other consumers.

**Review by isacikgoz — APPROVED** [review]
Nice 🚀

**Comment by hanzei** [comment]
@agnivade Do you also want to clean up other customer pointer implementations like the one in https://github.com/mattermost/mattermost/blob/f2907454960e9f05f5b3c3f3cb2864e71932cff2/server/channels/utils/merge_test.go#L1657-L1670 as part of this PR?

**Comment by agnivade** [comment]
I have a branch which takes care of this. But it's a LOT of changes. I just didn't want to mess up this PR more than needed. Perhaps I'll open an HW ticket to take care of that.

**Pull request merged**